android/src/main/java/com/reactnative/googlefit/HealthHistory.java

```diff
@@ -102,6 +102,47 @@ else if (dataReadResult.getDataSets().size() > 0) {
         return map;
     }
 
+    /**
+     * GLE added to allow us to aggregate heart rate data.
+     * It does the same as health history, but adds the aggregation.
+     * Note there are also some changes to the processDataSet method to allow for the aggregation.
+     */
+    public ReadableArray getAggregatedHeartRateHistory(long startTime, long endTime, int bucketInterval, String bucketUnit) {
+        DataReadRequest.Builder readRequestBuilder = new DataReadRequest.Builder()
+                .setTimeRange(startTime, endTime, TimeUnit.MILLISECONDS);
+
+        if (this.dataType == DataType.TYPE_HEART_RATE_BPM) {
+            readRequestBuilder
+                    .aggregate(this.dataType, DataType.AGGREGATE_HEART_RATE_SUMMARY)
+                    .bucketByTime(bucketInterval, HelperUtil.processBucketUnit(bucketUnit));
+        } else {
+            readRequestBuilder.read(this.dataType);
+        }
+
+        DataReadRequest readRequest = readRequestBuilder.build();
+
+        DataReadResult dataReadResult = Fitness.HistoryApi.readData(googleFitManager.getGoogleApiClient(), readRequest).await(1, TimeUnit.MINUTES);
+
+        WritableArray map = Arguments.createArray();
+
+        //Used for aggregated data
+        if (dataReadResult.getBuckets().size() > 0) {
+            for (Bucket bucket : dataReadResult.getBuckets()) {
```

Comment:
Worth storing the result of `dataReadResult.getBuckets()` into a variable to save calling it twice here?